forte/data/data_pack.py

```diff
@@ -585,31 +608,79 @@ def set_audio(
         self,
         audio: np.ndarray,
         sample_rate: int,
-        audio_payload_index: int = 0,
+        audio_payload_index: int = -1,
     ):
-        r"""Set the audio payload and sample rate of the :class:`~forte.data.data_pack.DataPack`
-        object.
+        r"""
+        Set audio for AudioPayload at a specified index or add a new AudioPayload in the DataPack.
+
+        Raises:
+            ValueError: raised when the audio payload index is out of range.
 
         Args:
             audio: A numpy array storing the audio waveform.
             sample_rate: An integer specifying the sample rate.
             audio_payload_index: the zero-based index of the AudioPayload
-                in this DataPack's AudioPayload entries. Defaults to 0.
+                in this DataPack's AudioPayload entries. Defaults to -1, and
+                it adds a new audio payload if there is no audio payload in the data pack.
```

Comment:
docstring different from `set_text`. The idea you have is interesting, probably want to make it more intuitive to users.